plugin/pkg/scheduler/core/equivalence_cache.go

```diff
@@ -18,29 +18,34 @@ package core
 
 import (
 	"hash/fnv"
-
-	"github.com/golang/groupcache/lru"
-
 	"sync"
 
+	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/kubernetes/pkg/api/v1"
 	hashutil "k8s.io/kubernetes/pkg/util/hash"
 	"k8s.io/kubernetes/plugin/pkg/scheduler/algorithm"
+
+	"github.com/golang/glog"
+	"github.com/golang/groupcache/lru"
 )
 
-// TODO(harryz) figure out the right number for this, 4096 may be too big
-const maxCacheEntries = 4096
+// we use predicate names as cache's key, its count is limited
+const maxCacheEntries = 100
```

Comment:
Yes, actually we did this in [the original PR](https://github.com/kubernetes/kubernetes/pull/30844/files#diff-f32e6323e72cadaaa2440fc35266f980R82), while after discussed with wojtek we all agreed to choose invalidation by predicate key over invalidate entirely cache everytime. Since we assume invalidation happens frequently.